utils/i18nUtils.js

```diff
@@ -0,0 +1,118 @@
+import i18next from "i18next";
+import { initReactI18next } from "react-i18next";
+import resourcesToBackend from "i18next-resources-to-backend";
+import { match } from "@formatjs/intl-localematcher";
+import { shouldPolyfill } from "@formatjs/intl-locale/should-polyfill";
+
+// Update this to the default locale for the app
+const DEFAULT_LOCALE = "en";
+// Update this to the supported locales contained in the `locales` folder
+const SUPPORTED_APP_LOCALES = ["en", "fr"];
+
+// TODO: Export polaris supported locales from the Polaris package
+const SUPPORTED_POLARIS_LOCALES = [
+  "cs",
+  "da",
+  "de",
+  "en",
+  "es",
+  "fi",
+  "fr",
+  "it",
+  "ja",
+  "ko",
+  "nb",
+  "nl",
+  "pl",
+  "pt-BR",
+  "pt-PT",
+  "sv",
+  "th",
+  "tr",
+  "vi",
+  "zh-CN",
+  "zh-TW",
+];
+
+let _userLocale, _polarisTranslations;
+
+export function getUserLocale() {
+  if (_userLocale) {
+    return _userLocale;
+  }
+  const url = new URL(window.location.href);
+  const locale = url.searchParams.get("locale") || DEFAULT_LOCALE;
+  _userLocale = match([locale], SUPPORTED_APP_LOCALES, DEFAULT_LOCALE);
+  return _userLocale;
+}
+
+export function getPolarisTranslations() {
+  return _polarisTranslations;
+}
+
+export async function initI18n() {
+  if (shouldPolyfill()) {
```

Comment:
The ponyfill [`intl-localematcher`](https://formatjs.io/docs/polyfills/intl-localematcher/) relies on [`Intl.getCanonicalLocales`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/getCanonicalLocales) and [`Intl.Locale`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale), or their polyfills.

`Intl.Locale` lists [browser compatibility](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale#browser_compatibility) with Safari 14 on iOS, and we currently support Safari 12.2 on iOS for POS web views. Therefore, we need to provide a polyfill for `Intl.Locale` in those cases.

I think that it would be better to dynamically import the polyfill in the few cases where it is needed, rather than always import it and add to the bundle size of the application.

The bundle size with the `Intl.Locale` polyfill is 758.65 KiB / gzip: 212.45 KiB:
![Uploading Screenshot 2023-04-14 at 9.49.11 AM.png…]()

The bundle size without the `Intl.Locale` polyfill is 587.56 KiB / gzip: 159.01 KiB.
The polyfill on its own is 171.59 Kib / gzip: 53.81 KiB:
<img width="489" alt="Screenshot 2023-04-14 at 9 49 41 AM" src="https://user-images.githubusercontent.com/1530553/232358164-158c5eeb-79c0-4a26-bda6-bf99b34ff0e6.png">